<h1>Importing Libraries</h1>

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

<h1>Importing Dataset</h1>

In [2]:
ibon = pd.read_csv('dataset\\penguins.csv')

<h1>Basic Checking</h1>

In [3]:
ibon

rowid    species     island  bill_length_mm  bill_depth_mm  \
0        1     Adelie  Torgersen            39.1           18.7   
1        2     Adelie  Torgersen            39.5           17.4   
2        3     Adelie  Torgersen            40.3           18.0   
3        4     Adelie  Torgersen             NaN            NaN   
4        5     Adelie  Torgersen            36.7           19.3   
..     ...        ...        ...             ...            ...   
339    340  Chinstrap      Dream            55.8           19.8   
340    341  Chinstrap      Dream            43.5           18.1   
341    342  Chinstrap      Dream            49.6           18.2   
342    343  Chinstrap      Dream            50.8           19.0   
343    344  Chinstrap      Dream            50.2           18.7   

     flipper_length_mm  body_mass_g     sex  year  
0                181.0       3750.0    male  2007  
1                186.0       3800.0  female  2007  
2                195.0       3250.0  female  2007  
3                  NaN          NaN     NaN  2007  
4                193.0       3450.0  female  2007  
..                 ...          ...     ...   ...  
339              207.0       4000.0    male  2009  
340              202.0       3400.0  female  2009  
341              193.0       3775.0    male  2009  
342              210.0       4100.0    male  2009  
343              198.0       3775.0  female  2009  

[344 rows x 9 columns]

<h1>Creating new column.</h1>

In [4]:
ibon['weight_kg'] = ibon['body_mass_g'] / 1000

In [5]:
ibon

rowid    species     island  bill_length_mm  bill_depth_mm  \
0        1     Adelie  Torgersen            39.1           18.7   
1        2     Adelie  Torgersen            39.5           17.4   
2        3     Adelie  Torgersen            40.3           18.0   
3        4     Adelie  Torgersen             NaN            NaN   
4        5     Adelie  Torgersen            36.7           19.3   
..     ...        ...        ...             ...            ...   
339    340  Chinstrap      Dream            55.8           19.8   
340    341  Chinstrap      Dream            43.5           18.1   
341    342  Chinstrap      Dream            49.6           18.2   
342    343  Chinstrap      Dream            50.8           19.0   
343    344  Chinstrap      Dream            50.2           18.7   

     flipper_length_mm  body_mass_g     sex  year  weight_kg  
0                181.0       3750.0    male  2007      3.750  
1                186.0       3800.0  female  2007      3.800  
2                195.0       3250.0  female  2007      3.250  
3                  NaN          NaN     NaN  2007        NaN  
4                193.0       3450.0  female  2007      3.450  
..                 ...          ...     ...   ...        ...  
339              207.0       4000.0    male  2009      4.000  
340              202.0       3400.0  female  2009      3.400  
341              193.0       3775.0    male  2009      3.775  
342              210.0       4100.0    male  2009      4.100  
343              198.0       3775.0  female  2009      3.775  

[344 rows x 10 columns]

In [6]:
#Rounding the value a table using the np.round(float, decimal_places) fucntion

In [7]:
ibon['weight_kg'] = np.round(ibon['body_mass_g'] / 1000, 1)
ibon

rowid    species     island  bill_length_mm  bill_depth_mm  \
0        1     Adelie  Torgersen            39.1           18.7   
1        2     Adelie  Torgersen            39.5           17.4   
2        3     Adelie  Torgersen            40.3           18.0   
3        4     Adelie  Torgersen             NaN            NaN   
4        5     Adelie  Torgersen            36.7           19.3   
..     ...        ...        ...             ...            ...   
339    340  Chinstrap      Dream            55.8           19.8   
340    341  Chinstrap      Dream            43.5           18.1   
341    342  Chinstrap      Dream            49.6           18.2   
342    343  Chinstrap      Dream            50.8           19.0   
343    344  Chinstrap      Dream            50.2           18.7   

     flipper_length_mm  body_mass_g     sex  year  weight_kg  
0                181.0       3750.0    male  2007        3.8  
1                186.0       3800.0  female  2007        3.8  
2                195.0       3250.0  female  2007        3.2  
3                  NaN          NaN     NaN  2007        NaN  
4                193.0       3450.0  female  2007        3.4  
..                 ...          ...     ...   ...        ...  
339              207.0       4000.0    male  2009        4.0  
340              202.0       3400.0  female  2009        3.4  
341              193.0       3775.0    male  2009        3.8  
342              210.0       4100.0    male  2009        4.1  
343              198.0       3775.0  female  2009        3.8  

[344 rows x 10 columns]

<h1>Getting the mean, mode, median</h1>

In [8]:
ibon['body_mass_g'].mean()

4201.754385964912

In [9]:
ibon['body_mass_g'].mode()

0    3800.0
Name: body_mass_g, dtype: float64

In [10]:
ibon['body_mass_g'].median()

4050.0

<h1>Checking the null values</h1>

In [11]:
ibon.isnull()

rowid  species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    False    False   False           False          False              False   
1    False    False   False           False          False              False   
2    False    False   False           False          False              False   
3    False    False   False            True           True               True   
4    False    False   False           False          False              False   
..     ...      ...     ...             ...            ...                ...   
339  False    False   False           False          False              False   
340  False    False   False           False          False              False   
341  False    False   False           False          False              False   
342  False    False   False           False          False              False   
343  False    False   False           False          False              False   

     body_mass_g    sex   year  weight_kg  
0          False  False  False      False  
1          False  False  False      False  
2          False  False  False      False  
3           True   True  False       True  
4          False  False  False      False  
..           ...    ...    ...        ...  
339        False  False  False      False  
340        False  False  False      False  
341        False  False  False      False  
342        False  False  False      False  
343        False  False  False      False  

[344 rows x 10 columns]

In [12]:
#Checking the null value but summarized.

In [13]:
ibon.isnull().sum()

rowid                 0
species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
weight_kg             2
dtype: int64

<h1>Filling up the null value</h1>

In [14]:
ibon['bill_length_mm'] = ibon['bill_length_mm'].fillna(ibon['bill_length_mm'].mean())

In [15]:
ibon['bill_depth_mm'] = ibon['bill_depth_mm'].fillna(ibon['bill_depth_mm'].mean())

In [16]:
ibon['flipper_length_mm'] = ibon['flipper_length_mm'].fillna(ibon['flipper_length_mm'].mean())

In [17]:
ibon['body_mass_g'] = ibon['body_mass_g'].fillna(ibon['body_mass_g'].mean())

In [18]:
ibon['sex'] = ibon['sex'].fillna(ibon['sex'].mode()[0])

In [2]:
ibon['sex'].mode()

NameError: name 'ibon' is not defined

In [19]:
ibon['weight_kg'] = ibon['weight_kg'].fillna(ibon['weight_kg'].mean())

In [20]:
ibon.isnull().sum()

rowid                0
species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
weight_kg            0
dtype: int64

<h1>Converting columns category</h1>

In [21]:
ibon['species'] = ibon['species'].astype('category')

In [22]:
ibon['island'] = ibon['island'].astype('category')

In [23]:
ibon['sex'] = ibon['sex'].astype('category')

In [24]:
ibon.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 10 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   rowid              344 non-null    int64   
 1   species            344 non-null    category
 2   island             344 non-null    category
 3   bill_length_mm     344 non-null    float64 
 4   bill_depth_mm      344 non-null    float64 
 5   flipper_length_mm  344 non-null    float64 
 6   body_mass_g        344 non-null    float64 
 7   sex                344 non-null    category
 8   year               344 non-null    int64   
 9   weight_kg          344 non-null    float64 
dtypes: category(3), float64(5), int64(2)
memory usage: 20.3 KB


In [25]:
ibon['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

<h1>Filtering data in specific data.</h1>

In [26]:
ibon[ibon['species'] == 'Adelie']

rowid species     island  bill_length_mm  bill_depth_mm  \
0        1  Adelie  Torgersen        39.10000       18.70000   
1        2  Adelie  Torgersen        39.50000       17.40000   
2        3  Adelie  Torgersen        40.30000       18.00000   
3        4  Adelie  Torgersen        43.92193       17.15117   
4        5  Adelie  Torgersen        36.70000       19.30000   
..     ...     ...        ...             ...            ...   
147    148  Adelie      Dream        36.60000       18.40000   
148    149  Adelie      Dream        36.00000       17.80000   
149    150  Adelie      Dream        37.80000       18.10000   
150    151  Adelie      Dream        36.00000       17.10000   
151    152  Adelie      Dream        41.50000       18.50000   

     flipper_length_mm  body_mass_g     sex  year  weight_kg  
0           181.000000  3750.000000    male  2007   3.800000  
1           186.000000  3800.000000  female  2007   3.800000  
2           195.000000  3250.000000  female  2007   3.200000  
3           200.915205  4201.754386    male  2007   4.202047  
4           193.000000  3450.000000  female  2007   3.400000  
..                 ...          ...     ...   ...        ...  
147         184.000000  3475.000000  female  2009   3.500000  
148         195.000000  3450.000000  female  2009   3.400000  
149         193.000000  3750.000000    male  2009   3.800000  
150         187.000000  3700.000000  female  2009   3.700000  
151         201.000000  4000.000000    male  2009   4.000000  

[152 rows x 10 columns]

In [27]:
ibon[(ibon['species'] == 'Adelie') & (ibon['island'] == 'Biscoe')]

rowid species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
20      21  Adelie  Biscoe            37.8           18.3              174.0   
21      22  Adelie  Biscoe            37.7           18.7              180.0   
22      23  Adelie  Biscoe            35.9           19.2              189.0   
23      24  Adelie  Biscoe            38.2           18.1              185.0   
24      25  Adelie  Biscoe            38.8           17.2              180.0   
25      26  Adelie  Biscoe            35.3           18.9              187.0   
26      27  Adelie  Biscoe            40.6           18.6              183.0   
27      28  Adelie  Biscoe            40.5           17.9              187.0   
28      29  Adelie  Biscoe            37.9           18.6              172.0   
29      30  Adelie  Biscoe            40.5           18.9              180.0   
50      51  Adelie  Biscoe            39.6           17.7              186.0   
51      52  Adelie  Biscoe            40.1           18.9              188.0   
52      53  Adelie  Biscoe            35.0           17.9              190.0   
53      54  Adelie  Biscoe            42.0           19.5              200.0   
54      55  Adelie  Biscoe            34.5           18.1              187.0   
55      56  Adelie  Biscoe            41.4           18.6              191.0   
56      57  Adelie  Biscoe            39.0           17.5              186.0   
57      58  Adelie  Biscoe            40.6           18.8              193.0   
58      59  Adelie  Biscoe            36.5           16.6              181.0   
59      60  Adelie  Biscoe            37.6           19.1              194.0   
60      61  Adelie  Biscoe            35.7           16.9              185.0   
61      62  Adelie  Biscoe            41.3           21.1              195.0   
62      63  Adelie  Biscoe            37.6           17.0              185.0   
63      64  Adelie  Biscoe            41.1           18.2              192.0   
64      65  Adelie  Biscoe            36.4           17.1              184.0   
65      66  Adelie  Biscoe            41.6           18.0              192.0   
66      67  Adelie  Biscoe            35.5           16.2              195.0   
67      68  Adelie  Biscoe            41.1           19.1              188.0   
100    101  Adelie  Biscoe            35.0           17.9              192.0   
101    102  Adelie  Biscoe            41.0           20.0              203.0   
102    103  Adelie  Biscoe            37.7           16.0              183.0   
103    104  Adelie  Biscoe            37.8           20.0              190.0   
104    105  Adelie  Biscoe            37.9           18.6              193.0   
105    106  Adelie  Biscoe            39.7           18.9              184.0   
106    107  Adelie  Biscoe            38.6           17.2              199.0   
107    108  Adelie  Biscoe            38.2           20.0              190.0   
108    109  Adelie  Biscoe            38.1           17.0              181.0   
109    110  Adelie  Biscoe            43.2           19.0              197.0   
110    111  Adelie  Biscoe            38.1           16.5              198.0   
111    112  Adelie  Biscoe            45.6           20.3              191.0   
112    113  Adelie  Biscoe            39.7           17.7              193.0   
113    114  Adelie  Biscoe            42.2           19.5              197.0   
114    115  Adelie  Biscoe            39.6           20.7              191.0   
115    116  Adelie  Biscoe            42.7           18.3              196.0   

     body_mass_g     sex  year  weight_kg  
20        3400.0  female  2007        3.4  
21        3600.0    male  2007        3.6  
22        3800.0  female  2007        3.8  
23        3950.0    male  2007        4.0  
24        3800.0    male  2007        3.8  
25        3800.0  female  2007        3.8  
26        3550.0    male  2007        3.6  
27        3200.0  female  2007        3.2  
28      

In [28]:
ibon[(ibon['species'] == 'Chinstrap') & (ibon['island'] == 'Biscoe')].info()

<class 'pandas.core.frame.DataFrame'>
Index: 0 entries
Data columns (total 10 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   rowid              0 non-null      int64   
 1   species            0 non-null      category
 2   island             0 non-null      category
 3   bill_length_mm     0 non-null      float64 
 4   bill_depth_mm      0 non-null      float64 
 5   flipper_length_mm  0 non-null      float64 
 6   body_mass_g        0 non-null      float64 
 7   sex                0 non-null      category
 8   year               0 non-null      int64   
 9   weight_kg          0 non-null      float64 
dtypes: category(3), float64(5), int64(2)
memory usage: 388.0 bytes


In [29]:
ibon[(ibon['species'] == 'Chinstrap') & (ibon['island'] == 'Dream')].info()

<class 'pandas.core.frame.DataFrame'>
Index: 68 entries, 276 to 343
Data columns (total 10 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   rowid              68 non-null     int64   
 1   species            68 non-null     category
 2   island             68 non-null     category
 3   bill_length_mm     68 non-null     float64 
 4   bill_depth_mm      68 non-null     float64 
 5   flipper_length_mm  68 non-null     float64 
 6   body_mass_g        68 non-null     float64 
 7   sex                68 non-null     category
 8   year               68 non-null     int64   
 9   weight_kg          68 non-null     float64 
dtypes: category(3), float64(5), int64(2)
memory usage: 4.8 KB


In [30]:
ibon[(ibon['species'] == 'Chinstrap') & (ibon['island'] == 'Torgersen')].info()

<class 'pandas.core.frame.DataFrame'>
Index: 0 entries
Data columns (total 10 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   rowid              0 non-null      int64   
 1   species            0 non-null      category
 2   island             0 non-null      category
 3   bill_length_mm     0 non-null      float64 
 4   bill_depth_mm      0 non-null      float64 
 5   flipper_length_mm  0 non-null      float64 
 6   body_mass_g        0 non-null      float64 
 7   sex                0 non-null      category
 8   year               0 non-null      int64   
 9   weight_kg          0 non-null      float64 
dtypes: category(3), float64(5), int64(2)
memory usage: 388.0 bytes


In [31]:
print('>> Biscoe')
ibon[(ibon['species'] == 'Adelie') & (ibon['island'] == 'Biscoe')].info()
print('\n>> Dream')
ibon[(ibon['species'] == 'Adelie') & (ibon['island'] == 'Dream')].info()
print('\n>> Torgersen')
ibon[(ibon['species'] == 'Adelie') & (ibon['island'] == 'Torgersen')].info()

>> Biscoe
<class 'pandas.core.frame.DataFrame'>
Index: 44 entries, 20 to 115
Data columns (total 10 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   rowid              44 non-null     int64   
 1   species            44 non-null     category
 2   island             44 non-null     category
 3   bill_length_mm     44 non-null     float64 
 4   bill_depth_mm      44 non-null     float64 
 5   flipper_length_mm  44 non-null     float64 
 6   body_mass_g        44 non-null     float64 
 7   sex                44 non-null     category
 8   year               44 non-null     int64   
 9   weight_kg          44 non-null     float64 
dtypes: category(3), float64(5), int64(2)
memory usage: 3.3 KB

>> Dream
<class 'pandas.core.frame.DataFrame'>
Index: 56 entries, 30 to 151
Data columns (total 10 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   rowid              56 non-

In [32]:
print('>> Biscoe')
ibon_biscoe = ibon[(ibon['species'] == 'Adelie') & (ibon['island'] == 'Biscoe')].info()
print('\n>> Dream')
ibon_dream = ibon[(ibon['species'] == 'Adelie') & (ibon['island'] == 'Dream')].info()
print('\n>> Torgersen')
ibon_torgersen = ibon[(ibon['species'] == 'Adelie') & (ibon['island'] == 'Torgersen')].info()

>> Biscoe
<class 'pandas.core.frame.DataFrame'>
Index: 44 entries, 20 to 115
Data columns (total 10 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   rowid              44 non-null     int64   
 1   species            44 non-null     category
 2   island             44 non-null     category
 3   bill_length_mm     44 non-null     float64 
 4   bill_depth_mm      44 non-null     float64 
 5   flipper_length_mm  44 non-null     float64 
 6   body_mass_g        44 non-null     float64 
 7   sex                44 non-null     category
 8   year               44 non-null     int64   
 9   weight_kg          44 non-null     float64 
dtypes: category(3), float64(5), int64(2)
memory usage: 3.3 KB

>> Dream
<class 'pandas.core.frame.DataFrame'>
Index: 56 entries, 30 to 151
Data columns (total 10 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   rowid              56 non-

In [33]:
ibon_biscoe = ibon[(ibon['species'] == 'Adelie') & (ibon['island'] == 'Biscoe')]
ibon_dream = ibon[(ibon['species'] == 'Adelie') & (ibon['island'] == 'Dream')]
ibon_torgersen = ibon[(ibon['species'] == 'Adelie') & (ibon['island'] == 'Torgersen')]

In [34]:
print('>> Biscoe')
np.round(ibon_biscoe['weight_kg'].mean(), 1)

>> Biscoe


3.7

In [35]:
print('\n>> Dream')
np.round(ibon_dream['weight_kg'].mean(), 1)


>> Dream


3.7

In [36]:
print('\n>> Torgersen')
np.round(ibon_torgersen['weight_kg'].mean(), 1)


>> Torgersen


3.7

In [37]:
ibon

rowid    species     island  bill_length_mm  bill_depth_mm  \
0        1     Adelie  Torgersen        39.10000       18.70000   
1        2     Adelie  Torgersen        39.50000       17.40000   
2        3     Adelie  Torgersen        40.30000       18.00000   
3        4     Adelie  Torgersen        43.92193       17.15117   
4        5     Adelie  Torgersen        36.70000       19.30000   
..     ...        ...        ...             ...            ...   
339    340  Chinstrap      Dream        55.80000       19.80000   
340    341  Chinstrap      Dream        43.50000       18.10000   
341    342  Chinstrap      Dream        49.60000       18.20000   
342    343  Chinstrap      Dream        50.80000       19.00000   
343    344  Chinstrap      Dream        50.20000       18.70000   

     flipper_length_mm  body_mass_g     sex  year  weight_kg  
0           181.000000  3750.000000    male  2007   3.800000  
1           186.000000  3800.000000  female  2007   3.800000  
2           195.000000  3250.000000  female  2007   3.200000  
3           200.915205  4201.754386    male  2007   4.202047  
4           193.000000  3450.000000  female  2007   3.400000  
..                 ...          ...     ...   ...        ...  
339         207.000000  4000.000000    male  2009   4.000000  
340         202.000000  3400.000000  female  2009   3.400000  
341         193.000000  3775.000000    male  2009   3.800000  
342         210.000000  4100.000000    male  2009   4.100000  
343         198.000000  3775.000000  female  2009   3.800000  

[344 rows x 10 columns]

<h1>Residence of specie per island</h1>
<h1>INSIGHT #1</h1>

In [38]:
chin = ibon[ibon['species'] == 'Chinstrap']
chin['island'].value_counts()

island
Dream        68
Biscoe        0
Torgersen     0
Name: count, dtype: int64

<h1>There is 68 Chinstrap penguins in Dream and none to Biscoe and Torgersen</h1>

<h1>INSIGHT #2</h1>

In [39]:
gen = ibon[ibon['species'] == 'Gentoo']
gen['island'].value_counts()

island
Biscoe       124
Dream          0
Torgersen      0
Name: count, dtype: int64

<h1>There is 124 Gentoo penguins in Biscoe and none to Dream and Torgersen</h1>

<h1>INSIGHT #3</h1>

In [40]:
ade = ibon[ibon['species'] == 'Adelie']
ade['island'].value_counts()

island
Dream        56
Torgersen    52
Biscoe       44
Name: count, dtype: int64

<h1>There is 44 Gentoo penguins in Biscoe and 56 to Dream and 52 in Torgersen</h1>

<h1>INSIGHT #4</h1>
<h1>Average weight in kg of species Chinstrap in every island</h1>

In [41]:
chin_dream = ibon[(ibon['island'] == 'Dream') & (ibon['species'] == 'Chinstrap')]

In [42]:
np.round(chin_dream['weight_kg'].mean(), 1)

3.7

<h1>The average weight in kg of Chinstrap in Dream Island is 3.7</h1>

<h1>INSIGHT #5</h1>
<h1>Average weight in kg of species Adelie in every island</h1>

In [43]:
ade_biscoe = ibon[(ibon['island'] == 'Biscoe') & (ibon['species'] == 'Adelie')]
ade_torgersen = ibon[(ibon['island'] == 'Torgersen') & (ibon['species'] == 'Adelie')]
ade_dream = ibon[(ibon['island'] == 'Dream') & (ibon['species'] == 'Adelie')]

In [44]:
np.round(ade_biscoe['weight_kg'].mean(), 1)

3.7

In [45]:
np.round(ade_torgersen['weight_kg'].mean(), 1)

3.7

In [46]:
np.round(ade_dream['weight_kg'].mean(), 1)

3.7

<h1>The average weight in kg of Adelie in every Island is 3.7</h1>

<h1>INSIGHT #6</h1>
<h1>Average weight in kg of species Gentoo in every island</h1>

In [47]:
gen_biscoe = ibon[(ibon['island'] == 'Biscoe') & (ibon['species'] == 'Gentoo')]

In [48]:
np.round(gen_biscoe['weight_kg'].mean(), 1)

5.1

<h1>The average weight in kg of Gentoo in Biscoe Island is 5.1</h1>

<h1>INSIGHT #7</h1>
<h1>The heaviest specie.</h1>

In [49]:
gen_biscoe = ibon[(ibon['island'] == 'Biscoe') & (ibon['species'] == 'Gentoo')]
np.round(gen_biscoe['weight_kg'].mean(), 1)

5.1

<h1>The heaviest specie is Gentoo in Biscoe Island with average of 5.1kg.</h1>

<h1>INSIGHT #8</h1>
<h1>The use of .groupby function</h1>

In [50]:
ibon_insight8 = ibon.groupby(['species', 'island', 'sex']).count()
ibon_insight8

C:\Users\User\AppData\Local\Temp\ipykernel_5600\3372943492.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  ibon_insight8 = ibon.groupby(['species', 'island', 'sex']).count()


rowid  bill_length_mm  bill_depth_mm  \
species   island    sex                                            
Adelie    Biscoe    female     22              22             22   
                    male       22              22             22   
          Dream     female     27              27             27   
                    male       29              29             29   
          Torgersen female     24              24             24   
                    male       28              28             28   
Chinstrap Biscoe    female      0               0              0   
                    male        0               0              0   
          Dream     female     34              34             34   
                    male       34              34             34   
          Torgersen female      0               0              0   
                    male        0               0              0   
Gentoo    Biscoe    female     58              58             58   
                    male       66              66             66   
          Dream     female      0               0              0   
                    male        0               0              0   
          Torgersen female      0               0              0   
                    male        0               0              0   

                            flipper_length_mm  body_mass_g  year  weight_kg  
species   island    sex                                                      
Adelie    Biscoe    female                 22           22    22         22  
                    male                   22           22    22         22  
          Dream     female                 27           27    27         27  
                    male                   29           29    29         29  
          Torgersen female                 24           24    24         24  
                    male                   28           28    28         28  
Chinstrap Biscoe    female                  0            0     0          0  
                    male                    0            0     0          0  
          Dream     female                 34           34    34         34  
                    male                   34           34    34         34  
          Torgersen female                  0            0     0          0  
                    male                    0            0     0          0  
Gentoo    Biscoe    female                 58           58    58         58  
                    male                   66           66    66         66  
          Dream     female                  0            0     0          0  
                    male                    0            0     0          0  
          Torgersen female                  0            0     0          0  
                    male                    0            0     0          0

<h1>Shows the use of .groupby() function.</h1>

In [51]:
ibon_insight8 = ibon.groupby(['species', 'island', 'sex', 'year']).size()
ibon_insight8

C:\Users\User\AppData\Local\Temp\ipykernel_5600\2813697338.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  ibon_insight8 = ibon.groupby(['species', 'island', 'sex', 'year']).size()


species    island     sex     year
Adelie     Biscoe     female  2007     5
                              2008     9
                              2009     8
                      male    2007     5
                              2008     9
                              2009     8
           Dream      female  2007     9
                              2008     8
                              2009    10
                      male    2007    11
                              2008     8
                              2009    10
           Torgersen  female  2007     8
                              2008     8
                              2009     8
                      male    2007    12
                              2008     8
                              2009     8
Chinstrap  Biscoe     female  2007     0
                              2008     0
                              2009     0
                      male    2007     0
                              2008     0
                      

<h1>The use of .size() in group data, using .groupby()</h1>

<h1>INSIGHT #9</h1>
<h1>Which is hevier amongst the penguin specie, Male Or Famale?</h1>

In [52]:
ibon_insight8 = np.round(ibon.groupby(['species', 'sex', 'year'])['weight_kg'].mean(), 1)
ibon_insight8

C:\Users\User\AppData\Local\Temp\ipykernel_5600\2071395540.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  ibon_insight8 = np.round(ibon.groupby(['species', 'sex', 'year'])['weight_kg'].mean(), 1)


species    sex     year
Adelie     female  2007    3.4
                   2008    3.4
                   2009    3.3
           male    2007    4.0
                   2008    4.1
                   2009    4.0
Chinstrap  female  2007    3.6
                   2008    3.5
                   2009    3.5
           male    2007    3.8
                   2008    4.1
                   2009    3.9
Gentoo     female  2007    4.6
                   2008    4.6
                   2009    4.8
           male    2007    5.5
                   2008    5.4
                   2009    5.4
Name: weight_kg, dtype: float64

In [53]:
ibon_insight8.unstack()

year              2007  2008  2009
species   sex                     
Adelie    female   3.4   3.4   3.3
          male     4.0   4.1   4.0
Chinstrap female   3.6   3.5   3.5
          male     3.8   4.1   3.9
Gentoo    female   4.6   4.6   4.8
          male     5.5   5.4   5.4

<h1>The data frame shows the weight of both female and male per species in different time frame.</h1>

<h1>QUESTION: Correlation between Bill length and bill depth</h1>

In [54]:
b_depth_lenght = ibon['bill_length_mm'].corr(ibon['bill_depth_mm'])

In [55]:
b_depth_lenght

-0.23505287035553282

<h1>Insight no.10</h1>

<h1>There is no correlation between bill lenght and bill depth.</h1>

<h1>QUESTION 11: Correlation between Body mass g and bill length</h1>

In [56]:
b_mass_bill_length = ibon['body_mass_g'].corr(ibon['bill_length_mm'])

In [57]:
b_mass_bill_length

0.5951098244376305

<h1>Insight no. 11</h1>

<h1>There is a correlation between bill lenght and body mass in grams.</h1>

<h1>QUESTION 12: Correlation between Body mass g and flipper length</h1>

In [58]:
b_mass_flipper_length = ibon['body_mass_g'].corr(ibon['flipper_length_mm'])

In [59]:
b_mass_flipper_length

0.8712017673060114

<h1>Insight no. 12</h1>

<h1>There is a correlation between flipper lenght and body mass in grams.</h1>

<h1>QUESTION 13: Correlation between bill depth and flipper length</h1>

In [60]:
b_depth_flipper_length = ibon['bill_depth_mm'].corr(ibon['flipper_length_mm'])

In [61]:
b_depth_flipper_length

-0.5838512164654122

<h1>Insight no. 13</h1>

<h1>There is no correlation between flipper length and bill depth.</h1>

<h1>QUESTION 14: Correlation between bill depth and body mass</h1>

In [62]:
b_depth_body_mass = ibon['bill_depth_mm'].corr(ibon['body_mass_g'])

In [63]:
b_depth_body_mass

-0.4719156211860664

<h1>Insight no. 14</h1>

<h1>There is no correlation between body mass and bill depth.</h1>

<h1>QUESTION 15: Correlation between bill depth and bill length in different species.</h1>

In [64]:
adelie = ibon[ibon['species'] == 'Adelie']
chinstrap = ibon[ibon['species'] == 'Chinstrap']
gentoo = ibon[ibon['species'] == 'Gentoo']

In [65]:
b_mass_bill_length_adelie = adelie['bill_depth_mm'].corr(adelie['bill_length_mm'])
b_mass_bill_length_adelie

0.3731961190689369

In [66]:
b_mass_bill_length_chin = chinstrap['bill_depth_mm'].corr(chinstrap['bill_length_mm'])
b_mass_bill_length_chin

0.6535362081800428

In [67]:
b_mass_bill_length_gentoo = gentoo['bill_depth_mm'].corr(gentoo['bill_length_mm'])
b_mass_bill_length_gentoo

0.6071518201350262

<h1>Insight no. 15</h1>

<h1>There is no correlation between body mass and bill depth but only in the Adelie species, but for Chinstrap and Gentoo there is a correlation.</h1>

<h1>INTALLING ALL NECESSARY PACKAGES</h1>

<h1>Testing Correlation Using One Way Anova, Analysis of varience</h1>

In [1]:
from scipy.stats import f_oneway

In [73]:
ibon[ibon['sex'] == 'female']['body_mass_g']

1      3800.0
2      3250.0
4      3450.0
6      3625.0
12     3200.0
        ...  
335    3525.0
337    3650.0
338    3650.0
340    3400.0
343    3775.0
Name: body_mass_g, Length: 165, dtype: float64

In [74]:
ibon[ibon['sex'] == 'male']['body_mass_g']

0      3750.000000
3      4201.754386
5      3650.000000
7      4675.000000
8      3475.000000
          ...     
334    3800.000000
336    3950.000000
339    4000.000000
341    3775.000000
342    4100.000000
Name: body_mass_g, Length: 179, dtype: float64

<h1>INSIGHT no. 16</h1>

In [81]:
f_stats, p_value = f_oneway(ibon[ibon['sex'] == 'male']['body_mass_g'],
                            ibon[ibon['sex'] == 'female']['body_mass_g'])

In [83]:
print(p_value)

3.0226600487050253e-15


<h1>The is correlation between the body mass in grams of a male penguins to a female.</h1>